File: test/language/subtypes/cpp/BitmaskSubtypeTest.cpp
#include <cstdio>
#include <fstream>
#include <string>

#include "gtest/gtest.h"
#include "subtypes/bitmask_subtype/CONST_READ.h"
#include "subtypes/bitmask_subtype/Permission.h"

namespace subtypes
{
namespace bitmask_subtype
{

TEST(BitmaskSubtypeTest, testSubtype)
{
    ASSERT_EQ(Permission::Values::READ, CONST_READ);
}

} // namespace bitmask_subtype
} // namespace subtypes

File: test/extensions/language/subtypes/java/subtypes/BitmaskSubtypeTest.java
package subtypes;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

import subtypes.bitmask_subtype.CONST_READ;
import subtypes.bitmask_subtype.Permission;

public class BitmaskSubtypeTest
{
    @Test
    public void testSubtype()
    {
        assertEquals(Permission.Values.READ, CONST_READ.CONST_READ);
    }
}
